src/fmeval/eval_algorithms/prompt_stereotyping.py

```diff
@@ -56,95 +99,94 @@ class PromptStereotyping(EvalAlgorithmInterface):
+        util.require(
+            sent_more_log_probability <= 0,
+            "Log-probabilities cannot be positive values. You might have passed raw probabilities instead.",
+        )
+        sample = {
+            DatasetColumns.SENT_MORE_LOG_PROB.value.name: sent_more_log_probability,
+            DatasetColumns.SENT_LESS_LOG_PROB.value.name: sent_less_log_probability,
+        }
+        get_scores = PromptStereotypingScores()
+        output = get_scores(sample)
+        return [EvalScore(name=LOG_PROBABILITY_DIFFERENCE, value=output[LOG_PROBABILITY_DIFFERENCE])]
+
     def evaluate(
         self,
         model: Optional[ModelRunner] = None,
         dataset_config: Optional[DataConfig] = None,
         prompt_template: Optional[str] = None,
+        num_records: int = 100,
         save: bool = False,
-        num_records=100,
     ) -> List[EvalOutput]:
+        """Compute prompt stereotyping metrics on one or more datasets.
+
+        :param model: An instance of ModelRunner representing the model under evaluation.
+        :param dataset_config: Configures the single dataset used for evaluation.
+            If not provided, evaluation will use all of its supported built-in datasets.
+        :param prompt_template: A template used to generate prompts that are fed to the model.
+            If not provided, defaults will be used.
+        :param num_records: The number of records to be sampled randomly from the input dataset
+            used to perform the evaluation.
+        :param save: If set to true, prompt responses and scores will be saved to a file.
+            The path that this file is stored at is configured by `eval_results_path`.
+
+        :return: A list of EvalOutput objects.
         """
-        Evaluate the model on how stereotypical it's responses are.
-
-        :param model: An instance of ModelRunner that represents the model being evaluated
-        :param dataset_config: The config to load the dataset to use for evaluation. If not provided, model will be
-                               evaluated on all built-in datasets configured for this evaluation.
-        :param prompt_template: A template which can be used to generate prompts, optional, if not provided defaults
-            will be used.
-        :param save: If set to true, prompt responses and scores will be saved to file. The output is written to
-                     EvalAlgorithmInterface.EVAL_RESULTS_PATH
-        :param num_records: The number of records to be sampled randomly from the input dataset to perform the
-                            evaluation
-
-        :return: a list of EvalOutput objects. Current implementation returns only one score.
-        """
-        if dataset_config:
-            dataset_configs = [dataset_config]
-        else:
-            dataset_configs = [DATASET_CONFIGS[dataset_name] for dataset_name in EVAL_DATASETS[self.eval_name]]
-
+        dataset_configs = get_dataset_configs(dataset_config, self.eval_name)
```

Comment:
NTH ideally this should also fall in line with the genral util.run_pipeline or something very close to that. The difference here is that rather than having a GetModelOutput transfor we have GetLogProbabilities

Reply:
Yeah, right now it'll take too much refactoring of existing code just to get `evaluate_dataset` to play nicely with PS, so I will handle this in the future.